src/bylines/index.js

```diff
@@ -3,64 +3,520 @@
+	}, [] );
+
+	return (
+		<div
+			className="newspack-byline-textarea"
+			contentEditable="true"
+			dangerouslySetInnerHTML={ { __html: byline.current } }
+		/>
+	);
+};
+
+const TokenInlineBlock = ( { token, onInsert } ) => {
+	return (
+		<>
+			<span
+				className="components-form-token-field__token token-inline-block"
+				id={ 'token-button-' + token.id }
+			>
+				<span className="components-form-token-field__token-text">
+					{ token.name }
+				</span>
+				<Button
+					className="components-form-token-field__insert-token is-small has-icon token-inline-block__insert"
+					onClick={ () => {
+						onInsert.call();
+					} }
+				>
+					<Icon icon={ plus } />
+				</Button>
+			</span>
+		</>
+	);
+};
+
+const Tokens = ( { tokens, tokensInUse, insertToken, onRendered } ) => {
+	useEffect( () => {
+		// Notify parent that this component has rendered.
+		onRendered();
+	}, [ onRendered ] );
+
+	return (
+		<div className="tokens">
+			{ tokens.map(
+				token =>
+					! tokensInUse.includes( token.id ) && (
+						<TokenInlineBlock
+							key={ token.id }
+							token={ token }
+							onInsert={ () => insertToken( token ) }
+						/>
+					)
+			) }
+		</div>
+	);
+};
 
 const BylinesSettingsPanel = () => {
+	/** Set when child components DOM are ready */
+	const [ isBylineReady, setIsBylineReady ] = useState( false );
+	const [ isTokensdReady, setIsTokensReady ] = useState( false );
+
+	/** Tokens with authors assigned to the post */
+	const [ tokens, setTokens ] = useState( [] );
+
+	/** Tokens that are in use by the custom byline */
+	const [ tokensInUse, setTokensInUse ] = useState( [] );
+
+	/** Reference to document to add event listners */
+	const documentRef = useRef( document );
+
+	/** Current post data */
+	const { postId } = useSelect(
+		select => ( {
+			postId: select( 'core/editor' ).getCurrentPostId(),
+		} ),
+		[]
+	);
+
+	/** coAuthors fetched from co-Authors Plus */
+	const [ coAuthors, setCoAuthors ] = useState( [] );
+
+	const noticesDispatch = useDispatch( 'core/notices' );
+
 	const { editPost } = useDispatch( 'core/editor' );
-	const { getEditedPostAttribute } = useSelect( select => select( 'core/editor' ) );
-	const [ byline, setByline ] = useState( getEditedPostAttribute( 'meta' )[ newspackBylines.metaKeyByline ] || '' );
-	const [ isEnabled, setIsEnabled ] = useState( !! getEditedPostAttribute( 'meta' )[ newspackBylines.metaKeyActive ] );
-	// Update byline text in editor.
-	useEffect( () => {
-		prependBylineToContent( isEnabled ? byline : '' );
-	}, [ byline, isEnabled ] );
-	// Enabled toggle handler.
+
+	const { getEditedPostAttribute } = useSelect( select =>
+		select( 'core/editor' )
+	);
+
+	/** Fetch post author from core */
+	const { postAuthor } = useSelect( select => {
+		const { getUser, getUsers } = select( coreStore );
+		const _authorId = getEditedPostAttribute( 'author' );
+		const query = { ...AUTHORS_QUERY };
+
+		return {
+			authors: getUsers( query ),
```

Comment:
This is being fetched but not returned; is it necessary?

Reply:
You're right, it was a reminiscence of a tentative or something, removed it